internal/txapp/txapp.go

```diff
@@ -262,25 +298,25 @@ func (r *TxApp) Execute(ctx TxContext, tx *transactions.Transaction) *TxResponse
 }
 
 // Begin signals that a new block has begun.
-// It contains information on any validators whose power should be updated.
-func (r *TxApp) Begin(ctx context.Context) error {
+// It contains information on any validators whos power should be updated.
+func (r *TxApp) Begin(ctx context.Context) (sql.OuterTx, error) {
```

Comment:
Returning a tx from a method and expecting that the caller does NOT commit it seems extremely weird.

Reply:
If we have to return a tx to abci, you should return the interface sql.DB instead. This makes it clear that the caller does not have to manage the lifecycle. I still don't think we should return it though.